src/dotnet/commands/dotnet-restore/Program.cs

```diff
@@ -131,11 +131,12 @@ private static void RestoreTool(LibraryRange tooldep, RestoreTask restoreTask, b
             {
                 var tempPath = Path.Combine(tempRoot, Guid.NewGuid().ToString(), "bin");
 
-                RestoreToolToPath(tooldep, restoreTask.Arguments, tempPath, quiet);
-
-                CreateDepsInPackageCache(tooldep, tempPath);
-
-                PersistLockFile(tooldep, tempPath, restoreTask.ProjectDirectory);
+                var restoreSucceded = RestoreToolToPath(tooldep, restoreTask.Arguments, tempPath, quiet);
+                if (restoreSucceded)
+                {
```

Comment:
when we saw the NREs I hadn't noticed any Nuget errors. I'll keep an eye out for undiagnosable issues...